starknet_devnet/blueprints/postman.py

```diff
@@ -55,3 +56,30 @@ async def flush():
 
     result_dict = await state.starknet_wrapper.postman_flush()
     return jsonify(result_dict)
+
+
+@postman.route("/l1_to_l2", methods=["POST"])
+async def l1_to_l2():
+    """L1 to L2 message mock endpoint"""
+    request_json = request.json or {}
+    l1_contract_address = extract_hex_string(request_json, "l1_contract_address")
+    l2_contract_address = extract_hex_string(request_json, "l2_contract_address")
+    entry_point_selector = extract_hex_string(request_json, "entry_point_selector")
+    payload = extract_hex_string(request_json, "payload", to_int_array)
+    nonce = extract_hex_string(request_json, "nonce")
+
+    result = await state.starknet_wrapper.postman_l1_to_l2(
+        l1_contract_address=l1_contract_address,
+        l2_contract_address=l2_contract_address,
+        entry_point_selector=entry_point_selector,
+        payload=payload,
+        nonce=nonce,
+    )
+
+    try:
+        result_iterator = iter(result)
+        return jsonify({"execution_info_calldata": list(map(hex, result_iterator))})
+    except TypeError:
```

Comment:
The result will be empty when the transaction will not be executed properly (for example contract doesn't exists).

TypeError will be raised in that case because it's not `iter`, but seems that `[hex(r) for r in result]` will fix it.

Reply:
so in the case of `[hex(r) for r in result]` the problem is the same, I'll try to find a way to refactor it later with another problem which is 200/500 HTTP response code

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/73cb0b6a24cec193377f8eedc2e7ddaf90177a8a

Reply:
you cannot iterate on empty set - this is why

but now with `[hex(r) for r in result]` everything works as expected

Reply:
Ok, can't the except clause then be removed?

Reply:
fixed in https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/8f1e7bb41cb060144d981dd6b122638b828b103f